server/store/kvstore/legal_hold.go

```diff
@@ -86,10 +82,6 @@ func (kvs Impl) GetLegalHoldByID(id string) (*model.LegalHold, error) {
 }
 
 func (kvs Impl) UpdateLegalHold(lh, oldValue model.LegalHold) (*model.LegalHold, error) {
-	if err := lh.IsValidForCreate(); err != nil {
```

Comment:
I would feel safer if we also kept the store level validation because we call this method from the job layer as well. We only call `CreateLegalHold` from the api layer but maybe we should also keep that validation in the store layer to protect ourselves from future modifications?

Reply:
Let's create a ticket for this an address for subsequent release.  We are in the final stage of releasing this to several customers.

Reply:
Created #105